multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/MergeDescriptorsStep.java

```diff
@@ -42,11 +45,21 @@ protected StepPhase executeStep(ProcessContext context) {
                                                                                                  extensionDescriptors);
         context.setVariable(Variables.DEPLOYMENT_DESCRIPTOR, descriptor);
 
+        warnForUnknownParameters(descriptor);
+
         backupDeploymentDescriptor(context, descriptor);
         getStepLogger().debug(Messages.DESCRIPTORS_MERGED);
         return StepPhase.DONE;
     }
 
+    private void warnForUnknownParameters(DeploymentDescriptor descriptor) {
+        List<String> unknownParameters = supportedParametersChecker.getUnknownParameters(descriptor);
```

Comment:
to me, it seems more logical to initialize the supportedParametersChecker with the descriptor variable. Since it doesn't make sense to invoke .getUnknownParameters() with different descriptors for the same instance. Also - this would probably mean we don't need to inject it.

Reply:
Leaving as it is, as we agreed in our internal discussion.